vic/drivers/image/src/get_force_type.c

```diff
@@ -0,0 +1,148 @@
+    char                    ncvarname[MAXSTRING];
+    int                     type;
+
+    type = SKIP;
+    strcpy(ncvarname, "MISSING");
+
+    /** Initialize flgstr **/
+    strcpy(flgstr, "NULL");
+
+    if ((*field) >= (int) param_set.N_TYPES[file_num]) {
+        log_err("Too many variables defined for forcing file %i., was "
+                "expecting at most %zu and got %d", file_num + 1,
+                param_set.N_TYPES[file_num], *field);
+    }
+
+    sscanf(cmdstr, "%*s %s %s", optstr, ncvarname);
+
+    /***************************************
+       Get meteorological data forcing info
+    ***************************************/
+
+    /* type 0: air temperature [C] */
+    if (strcasecmp("AIR_TEMP", optstr) == 0) {
+        type = AIR_TEMP;
+    }
+    /* type 1: albedo [fraction] */
+    else if (strcasecmp("ALBEDO", optstr) == 0) {
+        type = ALBEDO;
+    }
+    /* type 2: atmospheric CO2 mixing ratio [ppm] */
+    else if (strcasecmp("CATM", optstr) == 0) {
+        type = CATM;
+    }
+    /* type 3: incoming channel flow [m3] */
+    else if (strcasecmp("CHANNEL_IN", optstr) == 0) {
+        type = CHANNEL_IN;
+    }
+    /* type 4: direct fraction of shortwave [fraction] */
+    else if (strcasecmp("FDIR", optstr) == 0) {
+        type = FDIR;
+    }
+    /* type 5: LAI [m2/m2] */
+    else if (strcasecmp("LAI_IN", optstr) == 0) {
+        type = LAI_IN;
+    }
+    /* type 6: incoming longwave radiation [W/m2] */
+    else if (strcasecmp("LONGWAVE",
+                        optstr) == 0 || strcasecmp("LWDOWN", optstr) == 0) {
+        type = LONGWAVE;
+    }
+    /* type 7: photosynthetically active radiation [uE/m2s] */
+    else if (strcasecmp("PAR", optstr) == 0) {
+        type = PAR;
+    }
+    /* type 8: precipitation [mm] */
+    else if (strcasecmp("PREC", optstr) == 0) {
+        type = PREC;
+    }
+    /* type 9: air pressure [kPa] */
+    else if (strcasecmp("PRESSURE", optstr) == 0) {
+        type = PRESSURE;
+    }
+    /* type 10: vapor pressure [kPa] */
+    else if (strcasecmp("VP", optstr) == 0) {
+        type = VP;
+    }
+    /* type 11: rainfall [mm] */
+    else if (strcasecmp("SHORTWAVE",
+                        optstr) == 0 || strcasecmp("SWDOWN", optstr) == 0) {
+        type = SHORTWAVE;
+    }
+    /* type 12: vegetation cover fraction */
+    else if (strcasecmp("VEGCOVER", optstr) == 0) {
+        type = VEGCOVER;
+    }
+    /* type 13: wind speed [m/s] */
+    else if (strcasecmp("WIND", optstr) == 0) {
+        type = WIND;
+    }
+    /* type 14: unused (blank) data */
+    else if (strcasecmp("SKIP", optstr) == 0) {
+        type = SKIP;
+    }
+    /** Undefined variable type **/
+    else {
+        log_err("Undefined forcing variable type %s in file %i.",
+                optstr, file_num + 1);
+    }
+
+    param_set.TYPE[type].SUPPLIED = file_num + 1;
+    param_set.FORCE_INDEX[file_num][(*field)] = type;
+
+    if (strcasecmp("MISSING", ncvarname) != 0) {
```

Comment:
It handles the case where `ncvarname` is not provided in the configuration file.  For example, 

```
FORCING1      (put the forcing path/prefix here  # Forcing file path and prefix, ending in "_"
FORCE_TYPE    AIR_TEMP 	   tas
FORCE_TYPE    PREC
```

Parsing of the last line would raise the error on line 141 because `PREC` does not have a variable name specified.